modules/openapi-generator/src/main/resources/Javascript/ApiClient.mustache

```diff
@@ -541,12 +541,15 @@
 {{/usePromises}}  };
 
 {{#emitJSDoc}}  /**
-   * Parses an ISO-8601 string representation of a date value.
+   * Parses an ISO-8601 string representation or epoch representation of a date value.
    * @param {String} str The date value as a string.
    * @returns {Date} The parsed date object.
    */
 {{/emitJSDoc}}  exports.parseDate = function(str) {
-    return new Date(str.replace(/T/i, ' '));
+    if (isNaN(str)) {
```

Comment:
> The Number.isNaN() method determines whether a value is NaN (Not-A-Number).

Ref: https://www.w3schools.com/jsref/jsref_isnan_number.asp

Did you mean to check if str is a number?

Reply:
No I meant to check if str is **not** a number. If it's not a number then we'll default to the existing logic otherwise it's a number that we can just pass into Date.

Reply:
OK. After reading https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/isNaN, I think we can go with your fix.